lib/puppet/report_sanitizer.rb

```diff
@@ -357,6 +357,8 @@ def initialize(
     end
 
     def sanitize(raw)
+      # Puppetserver seems to remove 'kind' even from format version 6 reports.
+      raw['kind'] ||= 'apply'
       sanitized = super
```

Comment:
The added field should go below the `super` call

Reply:
The field was removed from the report (although it should still be there for report format 6)  supposedly by the puppetserver. We have to set it before we call 'super' because that is where we check for it's presence.